File: backend/app/core/supabase_client.py
"""
Deux clients Supabase distincts :
- service_client : clé service_role, bypass la RLS. Réservé aux opérations backend
  qui font autorité (vérif RBAC, écritures orchestrées, Storage). NE JAMAIS exposer
  cette clé au frontend.
- anon_client : clé anon, respecte la RLS. Utile si le backend doit exécuter une
  requête "au nom" d'un utilisateur sans bypasser ses droits.
"""

from functools import lru_cache

from supabase import Client, create_client

from app.core.config import settings


@lru_cache
def get_service_client() -> Client:
    return create_client(settings.SUPABASE_URL, settings.SUPABASE_SERVICE_ROLE_KEY)


@lru_cache
def get_anon_client() -> Client:
    return create_client(settings.SUPABASE_URL, settings.SUPABASE_ANON_KEY)

File: frontend/src/lib/supabaseClient.ts
import { createClient } from "@supabase/supabase-js";

const supabaseUrl = import.meta.env.VITE_SUPABASE_URL;
const supabaseAnonKey = import.meta.env.VITE_SUPABASE_ANON_KEY;

if (!supabaseUrl || !supabaseAnonKey) {
  throw new Error(
    "VITE_SUPABASE_URL et VITE_SUPABASE_ANON_KEY doivent être définis dans frontend/.env"
  );
}

// Seule la clé "anon" est utilisée côté frontend — elle respecte la RLS.
// La clé "service_role" ne doit JAMAIS apparaître dans le code frontend.
export const supabase = createClient(supabaseUrl, supabaseAnonKey);
